### PR TITLE
Valid fake hashes & additional hash helper utilities

### PR DESCRIPTION
**Pull request by mattyg**
- add helper function `dhtLocationFrom32` to construct valid dht location bytes from core 32 bytes
- alter `fakeAgentPubKey`, `fakeEntryHash`, `fakeActionHash`, `fakeDnaHash` to produce valid hashes, and to optionally take a param `coreByte` which will be repeated for all core 32 bytes
- add helper functions to slice up a full hash: `sliceDhtLocation`, `sliceCore32`, `sliceHashType`
- add helper functions to construct a full hash: `hashFrom32AndType`
- add helper object `hashTypePrefix` to store hash type label -> hash type bytes
- adds dependency `@bitgo/blake2b`
- Tests for all functions and new features

Feel free to rename functions as you please.

### CODE REVIEW

**Comment by mattyg** [comment]
@jost-s tagging you to request review since I can't don't have permission to assign you.

File: src/utils/hash-parts.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import blake2b from "@bitgo/blake2b";
+
+/**
+ * Hash type labels and their 3 byte values (forming the first 3 bytes of hash)
+ *
+ * From https://github.com/holochain/holochain/blob/develop/crates/holo_hash/src/hash_type/primitive.rs
+ *
+ * @public
+ */
+export const hashTypePrefix = {
```
Comment (jost-s) [inline src/utils/hash-parts.ts]:
```suggestion
export const HASH_TYPE_PREFIX = {
```

File: test/e2e/utils.ts [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+        132, 32, 36, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 126, 207, 206, 190,
+      ]),
+      "fakeAgentPubKey with coreByte set generates determanistic valid hash that decodes to AgentPubKey"
+    );
+  })
+);
+
+test(
+  "fakeEntryHash generates determanistic valid EntryHash when coreByte defined",
```
Comment (jost-s) [inline test/e2e/utils.ts]:
```suggestion
  "fakeEntryHash generates deterministic valid EntryHash when coreByte defined",
```

File: test/e2e/utils.ts [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+    t.deepEqual(
+      response,
+      Buffer.from(fakeHash),
+      "fakeDnaHash generates valid hash that decodes to DnaHash"
+    );
+  })
+);
+
+test(
+  "fakeAgentPubKey generates determanistic valid AgentPubKey when coreByte defined",
```
Comment (jost-s) [inline test/e2e/utils.ts]:
```suggestion
  "fakeAgentPubKey generates deterministic valid AgentPubKey when coreByte defined",
```

**Review by jost-s — APPROVED** [review]
LGTM! Thanks for the complete feature

File: src/utils/hash-parts.ts (hunk 1 above)
Comment (mattyg) [reply src/utils/hash-parts.ts]: done

File: test/e2e/utils.ts (hunk 2 above)
Comment (mattyg) [reply test/e2e/utils.ts]: done

File: test/e2e/utils.ts (hunk 3 above)
Comment (mattyg) [reply test/e2e/utils.ts]: fixed throughout the file

**Comment by mattyg** [comment]
@jost-s should be all set -- I don't have permission to merge

**Comment by jost-s** [comment]
Great, and thanks for the changelog!
I know about [Matthew's holo hash package](https://github.com/spartan-holochain-counsel/holo-hash-js/tree/master) and I want to incorporate that into the JS client, but AFAIK it doesn't generate fake hashes, so your work is not a duplicate.

**Pull request merged**